File: leetcode/0006-palindrome-number/isPalindrome.ts
function isPalindrome(x: number): boolean {
  if (x < 0) {
    return false;
  }
  let length = 1;
  while (Math.pow(10, length) <= x) {
    length++;
  }

  for (let left = length, right = 1; left > right; left--, right++) {
    const rightNumber = Math.floor(
      (x % Math.pow(10, right)) / Math.pow(10, right - 1)
    );

    const leftNumber = Math.floor(x / Math.pow(10, left - 1)) % 10;

    if (rightNumber !== leftNumber) {
      return false;
    }
  }

  return true;
}

export default isPalindrome;

File: leetcode/0006-palindrome-number/isPalindrome.go
package leetcode

import "math"

func isPalindrome(x int) bool {
	if x < 0 {
		return false
	}

	var length = 1

	for x >= int(math.Pow(10, float64(length))) {
		length++
	}
	var (
		left  = length
		right = 1
	)

	for left > right {
		var leftNumber = x / int(math.Pow(10, float64(left-1))) % 10

		var rightNumber = x % int(math.Pow(10, float64(right))) / int(math.Pow(10, float64(right-1)))

		if leftNumber != rightNumber {
			return false
		}

		left--
		right++
	}

	return true
}
